patchtools/patch.py

```diff
@@ -152,15 +160,21 @@ def from_email(self, msg):
         if not self.repo:
             f = self.find_repo()
 
-        if self.repo in self.mainline_repo_list:
+        remote = self.get_commit_remote_name()
+        is_origin_commit = (remote == "origin" or remote == "undefined")
```

Comment:
I think you're close here, but we can generalize this.  You've added support to look up the name of the remote and the remote url -- why not ignore whether the remote is origin and instead compare it to the list of known upstream mainline URLs?